Add files via upload

Stanislav-prog opened the pull request.

Stanislav-prog committed:
Add files via upload

RuslanFrankov commented on css/null.css:
```diff
@@ -0,0 +1,32 @@
+*,*::before,*::after{
```
i wonder why this file is called `null`. Does is mean a basic set of styles for main html tags?

Stanislav-prog replied:
> i wonder why this file is called `null`. Does is mean a basic set of styles for main html tags?

this file is called 'null', because it makes nullification of basic set of styles in each browser. I've downloaded it from the Internet, there are many properties i don't know about, but it is really useful

RuslanFrankov commented on index.html:
```diff
@@ -0,0 +1,28 @@
+    <div class="wrapper">
+        <h1 class="title">Online Library</h1>
+        <form action="#" class="book-form">
+            <input type="text" class="book-form__input book-form__input--id" placeholder="ID" name="id" required>
```
I can't find style for `book-form__input--id` . Do you know if it is needed at all?

Stanislav-prog replied:
> I can't find style for `book-form__input--id` . Do you know if it is needed at all?

there's no style, but I made it, because I needed to get each element for variables `formId`, `formTitle`, etc... But I also created global `book-form__input` to style them all. Maybe there's a way to get each element without separate class names?

RuslanFrankov replied:
I think it's better to find element by id, see https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementById
but you should  make sure that all ids are unique

RuslanFrankov commented on index.html:
```diff
@@ -0,0 +1,28 @@
+            <input type="text" class="book-form__input book-form__input--year" placeholder="Edition year" name="edition_year" required>
+            <input type="text" class="book-form__input book-form__input--author" placeholder="Author" name="author" required>
+            <input type="text" class="book-form__input book-form__input--cost" placeholder="Cost" name="cost" required>
+            <input type="text" class="book-form__input book-form__input--copies" placeholder="Number of copies" name="number-of-copies" required>
```
would it be better to use type="number" for copies and cost fields ?

RuslanFrankov commented on index.html:
```diff
@@ -0,0 +1,28 @@
+        <form action="#" class="book-form">
+            <input type="text" class="book-form__input book-form__input--id" placeholder="ID" name="id" required>
+            <input type="text" class="book-form__input book-form__input--title" placeholder="Title" name="title" required>
+            <input type="text" class="book-form__input book-form__input--year" placeholder="Edition year" name="edition_year" required>
```
I think that year is a numeric value and we might want to change the type to number. What do you think?

RuslanFrankov commented on js/script.js:
```diff
@@ -0,0 +1,79 @@
+var books = [];
```
it seems that having `var` type of field is a bit dangerous here  as you may change the reference unintentionally later. `books` plays a role of a mock database, so the reference should be constant.

RuslanFrankov commented on js/script.js:
```diff
@@ -0,0 +1,79 @@
+var books = [];
+
+const form = document.querySelector('.book-form');
+const formId = document.querySelector('.book-form__input--id');
+const formTitle = document.querySelector('.book-form__input--title');
+const formEditionYear = document.querySelector('.book-form__input--year');
+const formAuthor = document.querySelector('.book-form__input--author');
+const formCost = document.querySelector('.book-form__input--cost');
+const formNumberOfCopies = document.querySelector('.book-form__input--copies');
```
good choice to select by class , however it might return multiple values for each element. One style class can be applied to multiple elements. How we can avoid this ? We need some attribute of the element that is unique across  entire document . Can you improve the approach of selecting elements ?

RuslanFrankov commented on js/script.js:
```diff
@@ -0,0 +1,79 @@
+const formCost = document.querySelector('.book-form__input--cost');
+const formNumberOfCopies = document.querySelector('.book-form__input--copies');
+
+form.addEventListener('submit', (e) => {
```
in case of multiple submit elements in a form this function will be triggered on all of the buttons. Can you create a  function for handling clicks on specific button ?

RuslanFrankov commented on js/script.js:
```diff
@@ -0,0 +1,79 @@
+        messageElement.textContent = '';
+    }, 3000)
+
+    function createTable(books){
```
i think that there is no sense in keeping this function inside addEventListener callback function. It's better to move it to the outer level of the script.

RuslanFrankov commented on js/script.js:
```diff
@@ -0,0 +1,79 @@
+
+        table.appendChild(headerRow);
+
+        books.forEach(book => {
```
it seems a bit of overhead to recreate table each time a new book is added to the list. Can you redesign the logic so that just one row is appended to the table ?

Stanislav-prog committed:
Add files via upload

Updated null.css
Removed styles for each `input` element, added `id` instead. Commited changes in JS correspondingly
Changed types to `number`
Changed keyword `var` to `const`
Added button to reset a form

Stanislav-prog committed:
Add files via upload

Created new function `addBook()` to trigger `onclick` in order to prevent cases when there are multiple `submit` buttons

Stanislav-prog committed:
Add files via upload
